system/lib/wasmfs/wasmfs.cpp

```diff
@@ -36,6 +36,11 @@ void _wasmfs_get_preloaded_path_name(int index, char* fileName);
 void _wasmfs_get_preloaded_child_path(int index, char* childName);
 }
 
+WasmFS::~WasmFS() {
+  StdoutFile::getSingleton()->flush();
+  StderrFile::getSingleton()->flush();
```

Comment:
Would it make sense to put this logic in the destructors of the singletons themselves rather than in the WasmFS destructor?

Reply:
Interesting idea, perhaps. But doing it all in one destructor is a way to get the right ordering - we need to flush libc first, then these. So for now I think it needs to be done as it is.

Reply:
Yes, I kind of like doing it all from here... I don't think we want to rely too much on the ordering of other destructors.    IIRC this desctrutor we know will be run very late because its constructor is scheduled explicitly to be very early, right?

Reply:
The constructor is early, yeah. Does the toolchain then emit the destructor late? That would be great if so!

Reply:
Yes, static constructors and destructors are run in a first in last out order.    In fact, what I believe happens under the hood is that the constructor registers its own destructor in a FILO dtor queue (using `__cxa_atexit`)